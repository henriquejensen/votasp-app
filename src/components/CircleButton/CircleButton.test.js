import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import CircleButton from './CircleButton';

describe('CircleButton', () => {
  it('should match a snapshot', () => {
    const onClick = jest.fn();
    const content = 'Hello World';
    const Component = shallow(( <CircleButton onClick={onClick} content={content} /> ));
    expect(toJson(Component)).toMatchSnapshot();
    Component.props().onClick();
    expect(onClick).toHaveBeenCalled();
  })

  it('should render correctly wehn onClick is not passed', () => {
    const onClick = jest.fn();
    const content = 'Hello World';
    const Component = shallow(( <CircleButton content={content} /> ));
    expect(toJson(Component)).toMatchSnapshot();
    Component.props().onClick();
    expect(onClick).not.toHaveBeenCalled();
  })
})
