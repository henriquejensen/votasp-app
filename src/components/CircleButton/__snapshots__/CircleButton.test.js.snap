// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleButton should match a snapshot 1`] = `
<button
  className="circle-button"
  onClick={[MockFunction]}
>
  Hello World
</button>
`;

exports[`CircleButton should render correctly wehn onClick is not passed 1`] = `
<button
  className="circle-button"
  onClick={[Function]}
>
  Hello World
</button>
`;
